crates/typed-store/src/rocks/mod.rs

```diff
@@ -438,22 +439,17 @@ where
 }
 
 fn read_size_from_env(var_name: &str) -> Option<usize> {
-    match env::var(var_name) {
-        Ok(val) => match val.parse::<usize>() {
-            Ok(i) => Some(i),
-            Err(e) => {
-                info!(
-                    "Env var {} does not contain valid usize integer: {}",
-                    var_name, e
-                );
-                Option::None
-            }
-        },
-        Err(e) => {
-            info!("Env var {} is not set: {}", var_name, e);
-            Option::None
-        }
-    }
+    env::var(var_name)
+        .tap_err(|e| info!("Env var {} is not set: {}", var_name, e))
+        .ok()?
+        .parse::<usize>()
+        .tap_err(|e| {
+            info!(
+                "Env var {} does not contain valid usize integer: {}",
+                var_name, e
+            )
+        })
+        .ok()
```

Comment:
```suggestion
    env::var(var_name).and_then(|v|.parse::<usize>()
        .tap_err(|e| {
            info!(
                "Env var {} unset or does not contain valid usize integer: {}",
                var_name, e
            )
        }))
        .ok()
```

Reply:
It seems the signature of `Result<U, E>::and_then()` is 
```
pub fn and_then<U, F: FnOnce(T) -> Result<U, E>>(self, op: F) -> Result<U, E>
```
So the `Result` and the closure in `and_then()` must have the same error type. Here `env::var()` and `parse::<usize>()` have different error types.

Reply:
Yup, I'd expect a `map_err` to come in at some point. I trust you to find the best compromise.